src/components/Calculator.js

```diff
@@ -1,49 +1,70 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
+import calculate from '../logic/Calculate';
 
-function CalBtn(props) {
-  const { incrementNumber, bgColor } = props;
-  return (<button type="button" className={bgColor}>{incrementNumber}</button>);
-}
+function Calculator() {
+  const [calcData, setCalcData] = useState({
+    total: null,
+    next: null,
+    operation: null,
+  });
 
-CalBtn.propTypes = {
-  incrementNumber: PropTypes.number.isRequired,
-  bgColor: PropTypes.string.isRequired,
-};
+  function callActions(e) {
+    const newValueInput = calculate(calcData, e.target.value);
+    setCalcData(newValueInput);
+  }
+  function CalBtn(props) {
+    const {
+      incrementNumber, characterStr, bgColor,
+    } = props;
+
+    return (<input type="button" onClick={callActions} className={bgColor} value={incrementNumber || characterStr} />);
+  }
```

Comment:
**[OPTIONAL]**

- [ ] It would be nice if you add this CalBtn component in a different file and import it here to organize your code in different files :+1: